packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -0,0 +1,1005 @@
+																		setCommentEdit(
+																			false
+																		);
+																	} }
+																>
+																	{ __(
+																		'Update'
+																	) }
+																</Button>
+																<Button
+																	className="is-compact"
+																	onClick={ () => {
+																		setCommentEdit(
+																			false
+																		);
+																		setShowConfirmation(
+																			false
+																		);
+																	} }
+																>
+																	{ __(
+																		'Cancel'
+																	) }
+																</Button>
+															</HStack>
+														</VStack>
+													</>
+												) }
+											<RawHTML>
+												{ reply.content.rendered }
+											</RawHTML>
+										</VStack>
+									</HStack>
+
+									{ hasCommentReply &&
+										reply.id === showConfirmationTabId && (
+											<HStack
+												alignment="left"
+												spacing="3"
+												justify="flex-start"
+											>
+												<VStack
+													spacing="3"
+													className="editor-collab-sidebar__replyComment"
+												>
+													<TextareaControl
+														className="editor-collab-sidebar__replyComment__textarea"
+														value={
+															commentReply ?? ''
+														}
+														onChange={ (
+															value
+														) => {
+															setCommentReply(
+																value
+															);
+														} }
+													/>
+													<VStack
+														alignment="left"
+														spacing="3"
+														justify="flex-start"
+														onRequestClose={ () =>
+															setHasCommentReply(
+																false
+															)
+														}
+													>
+														<HStack
+															alignment="left"
+															spacing="3"
+															justify="flex-start"
+															className="editor-collab-sidebar__replybtn"
+														>
+															<Button
+																variant="primary"
+																onClick={ () => {
+																	confirmReplyComment(
+																		reply.id
+																	);
+																	setHasCommentReply(
+																		false
+																	);
+																} }
+																className="is-compact"
+															>
+																{ __(
+																	'Reply'
+																) }
+															</Button>
+															<Button
+																onClick={ () => {
+																	setHasCommentReply(
+																		false
+																	);
+																	setShowConfirmation(
+																		false
+																	);
+																} }
+																className="is-compact"
+															>
+																{ __(
+																	'Cancel'
+																) }
+															</Button>
+														</HStack>
+													</VStack>
+												</VStack>
+											</HStack>
+										) }
+
+									{ commentConfirmation &&
```

Comment:
Was about to comment the same. It feels like these file need a lot of refactoring, adding a level of abstraction for common patterns and componentizing the UI to avoid very deep nesting

Reply:
We have refactored the code and modularized it into components.